tests/admin_views/tests.py

```diff
@@ -4688,6 +4688,21 @@ def test_readonly_backwards_ref(self):
         response = self.client.get(reverse('admin:admin_views_topping_add'))
         self.assertEqual(response.status_code, 200)
 
+    def test_readonly_onetoone_backwards_ref(self):
+        """
+        Regression test for OneToOneField backward reference, when related field is read-only.
+        """
+        v1 = Villain.objects.create(name='Adam')
+        pl = Plot.objects.create(name='Test Plot', team_leader=v1, contact=v1)
+        PlotDetails.objects.create(details='Brand New Plot', plot=pl)
+        response = self.client.get(reverse('admin:admin_views_plotproxy_change', args=(pl.pk,)))
+        fieldset = list(response.context['adminform'])[0]
```

Comment:
It's ugly either way, but this way might make it slightly easier to debug if the test fails at some later point.
```python
plot_details_field = list(list(list(response.context['adminform'])[0])[3])[0]
self.assertEqual(plot_details_field.field['name'], 'plotdetails')
self.assertTrue(plot_details_field.is_readonly)
self.assertEqual(plot_details_field.contents(), 'Brand New Plot')
```
I guess it we used the pattern more widely, it might be worth some helper functions to make it easy to extract fields without using magic numbers in the indexing.